src/MerchantCenter/MerchantCenterService.php

```diff
@@ -95,15 +97,24 @@ public function is_google_connected(): bool {
 	/**
 	 * Whether we are able to sync data to the Merchant Center account.
 	 * Account must be connected and the URL we claimed with must match the site URL.
+	 * URL matches is stored in a transient to prevent it from being refetched in cases
+	 * where the site is unable to access account data.
 	 *
 	 * @since 1.13.0
 	 * @return boolean
 	 */
 	public function is_ready_for_syncing(): bool {
-		$claimed_url_hash = $this->container->get( Merchant::class )->get_claimed_url_hash();
-		$site_url_hash    = md5( $this->get_site_url() );
+		$transients  = $this->container->get( TransientsInterface::class );
```

Comment:
Minor: Let's add a comment to help out IDEs.

```suggestion
		/** @var TransientsInterface $transients */
		$transients  = $this->container->get( TransientsInterface::class );
```